docs/notifications.md

```diff
@@ -531,6 +537,37 @@ Example:
 }
 ```
 
+### `header_of_added_block` notification
```

Comment:
We need to add a couple of words to `## Ordering and persistence guarantees` section.